Kickstarter-iOS/Features/ReportProject/ReportProjectFormView.swift

```diff
@@ -60,8 +69,8 @@ struct ReportProjectFormView: View {
       .toolbar {
         ToolbarItem(placement: .navigationBarTrailing) {
           LoadingBarButtonItem(
-            saveEnabled: $saveEnabled,
-            saveTriggered: $saveTriggered,
+            saveEnabled: $viewModel.saveButtonEnabled,
+            saveTriggered: $viewModel.saveTriggered,
```

Comment:
So this is an interesting one - I'm not actually sure that it makes sense for a `triggered` action to be a binding like this. Right now, we're modeling `saveTriggered` as an event stream of `Bools`. For example, if you press the button, you'll get the following stream:
```
false - initial state
true - the button was pressed
false - we set it back to false once the loading is done
```
However, it's not really a _state_, it's more of just a thing that happens. It would make more sense to me to model it as a stream of voids, where only one event occurs when the button is pressed.

I actually like Apple's pattern here with the underlying View class `Button`, where it takes a closure that defines what happens when the button is pressed. This would let us follow more of our old input/output pattern that we used with UIKit:

```
LoadingBarButtonItem(
  saveEnabled: $saveEnabled,
  showLoading: $showLoading,
  titleText: Strings.Send()
) {
  viewModel.didPressSaveButton()
}
```

Reply:
I like the closure approach better, too!

Reply:
Cool, I'll throw myself a quick ticket to refactor this. Just to keep this PR a bit more svelte.